sim/core/item_swaps.go

```diff
@@ -68,71 +68,227 @@ func (character *Character) enableItemSwap(itemSwap *proto.ItemSwap, mhCritMulti
 		ohCritMultiplier:     ohCritMultiplier,
 		rangedCritMultiplier: rangedCritMultiplier,
 		slots:                slots,
+		originalEquip:        character.Equipment,
+		swapEquip:            swapItems,
 		unEquippedItems:      swapItems,
-		swapped:              false,
+		swapSet:              proto.APLActionItemSwap_Unknown,
+		initialized:          false,
 	}
 }
 
 func (swap *ItemSwap) initialize(character *Character) {
 	swap.character = character
 }
 
-func (character *Character) RegisterOnItemSwap(callback OnSwapItem) {
+func (character *Character) RegisterItemSwapCallback(slots []proto.ItemSlot, callback OnItemSwap) {
 	if character == nil || !character.ItemSwap.IsEnabled() {
 		return
 	}
 
```

Comment:
We should add a similar "infinite registration" check here like we do for Aura registration etc. in order to prevent accidental developer errors:
```
if (character.Env != nil) && character.Env.IsFinalized() {
    panic("Tried to add a new item swap callback in a finalized environment!")
}
```